packages/botonic-react/src/components/message.jsx

```diff
@@ -240,73 +217,70 @@ export const Message = props => {
-            bgcolor={getBgColor()}
-            color={isFromUser() ? COLORS.SOLID_WHITE : COLORS.SOLID_BLACK}
-            blobWidth={getThemeProperty('message.bot.blobWidth')}
-            blob={blob}
+        <>
+          <MessageContainer
+            isfromuser={isFromUser()}
             style={{
-              ...getMessageStyle(),
-              ...style,
+              ...getThemeProperty('message.style'),
             }}
-            {...otherProps}
           >
-            {markdown ? (
-              <BlobText
-                blob={blob}
-                dangerouslySetInnerHTML={{
-                  __html: renderMarkdown(textChildren),
+            {isFromBot() && BotMessageImage && (
+              <BotMessageImageContainer
+                style={{
+                  ...getThemeProperty('message.bot.imageStyle'),
+                  ...imageStyle,
                 }}
-                markdownstyle={getMarkdownStyle(
-                  getThemeProperty,
-                  isFromUser() ? COLORS.SEASHELL_WHITE : brandColor
-                )}
-              />
-            ) : (
-              <BlobText blob={blob}>{textChildren}</BlobText>
+              >
+                <img
+                  style={{ width: '100%' }}
+                  src={resolveImage(BotMessageImage)}
+                />
+              </BotMessageImageContainer>
             )}
-            {buttons}
-            {blob && hasBlobTick() && getBlobTick(6)}
-            {blob && hasBlobTick() && getBlobTick(5)}
-          </Blob>
-        </MessageContainer>
-        <TimestampContainer>
-          {timestampFormat && (
-            <TimestampText
-              isfromuser={isFromUser()}
+            <Blob
+              bgcolor={getBgColor()}
+              color={isFromUser() ? COLORS.SOLID_WHITE : COLORS.SOLID_BLACK}
+              blobWidth={getThemeProperty('message.bot.blobWidth')}
+              blob={blob}
               style={{
-                ...getThemeProperty('message.timestamps.style'),
+                ...getMessageStyle(),
+                ...style,
               }}
+              {...otherProps}
             >
-              {m.timestamp}
-            </TimestampText>
+              {markdown ? (
+                <BlobText
+                  blob={blob}
+                  dangerouslySetInnerHTML={{
+                    __html: renderMarkdown(textChildren),
+                  }}
+                  markdownstyle={getMarkdownStyle(
+                    getThemeProperty,
+                    isFromUser() ? COLORS.SEASHELL_WHITE : brandColor
+                  )}
+                />
+              ) : (
+                <BlobText blob={blob}>{textChildren}</BlobText>
+              )}
+              {buttons}
+              {blob && hasBlobTick() && getBlobTick(6)}
+              {blob && hasBlobTick() && getBlobTick(5)}
+            </Blob>
+          </MessageContainer>
+          {timestampsEnabled && (
+            <MessageTimestamp
```

Comment:
would it be possible to pass getThemeProperty to MessageTimestamp (or MessageTimestamp gets its from context) so that resolveMessageTimestamps is called from MessageTimestamp?
Then MessageTimestamp would be more autocontained and Message would be simpler